public/app/features/scenes/components/layout/SceneGridLayout.tsx

```diff
@@ -0,0 +1,403 @@
+              cols={GRID_COLUMN_COUNT}
+              rowHeight={GRID_CELL_HEIGHT}
+              draggableHandle={`.grid-drag-handle-${model.state.key}`}
+              // @ts-ignore: ignoring for now until we make the size type numbers-only
+              layout={width > 768 ? layout.lg : layout.sm}
+              onDragStop={model.onDragStop}
+              onResizeStop={model.onResizeStop}
+              onLayoutChange={model.onLayoutChange}
+              isBounded={true}
+            >
+              {children.map((child) => {
+                return (
+                  // Grid items are flex, to be able to render flex layouts inside
+                  <div key={child.state.key} style={{ display: 'flex' }}>
+                    <child.Component model={child} key={child.state.key} />
+                  </div>
+                );
+              })}
+            </ReactGridLayout>
+          </div>
+        );
+      }}
+    </AutoSizer>
+  );
+}
+
+interface SceneGridRowState extends SceneLayoutChildState {
+  title: string;
+  isCollapsible?: boolean;
+  isCollapsed?: boolean;
+  children: Array<SceneObject<SceneLayoutChildState>>;
+}
+
+export class SceneGridRow extends SceneObjectBase<SceneGridRowState> {
+  static Component = SceneGridRowRenderer;
+  private _originalHeight = 0;
+
+  constructor(
+    state: Omit<SceneGridRowState, 'size'> & { size: Pick<GridCellLayout, 'x' | 'y' | 'height'> & { width?: number } }
+  ) {
+    super({
+      isResizable: true,
+      isDraggable: true,
+      isCollapsible: true,
+      ...state,
+      size: {
+        ...state.size,
+        height: state.isCollapsed ? GRID_CELL_HEIGHT : state.size?.height || DEFAULT_ROW_HEIGHT,
+        width: state.size.width || GRID_COLUMN_COUNT,
+      },
+    });
+
+    this._originalHeight = parseInt(
+      (state.isCollapsed ? GRID_CELL_HEIGHT : state.size?.height || DEFAULT_ROW_HEIGHT).toString(),
+      10
+    );
+
+    this.subs = this.subscribe({
+      next: (state) => {
+        // Preserve the height of the row to be able to restore it when uncollapsing
+        if (
+          state.size &&
+          state.size.height &&
+          state.size.height !== this._originalHeight &&
+          state.size?.height !== GRID_CELL_HEIGHT &&
+          !state.isCollapsed
+        ) {
+          this._originalHeight = parseInt(state.size.height?.toString(), 10);
+        }
+      },
+    });
+  }
+
+  onCollapseToggle = () => {
+    if (!this.state.isCollapsible) {
+      return;
+    }
+    const layout = this.parent;
+
+    if (!layout || !(layout instanceof SceneGridLayout)) {
+      throw new Error('SceneGridRow must be a child of SceneGridLayout');
+    }
+
+    const { isCollapsed, size } = this.state;
+    if (!size) {
+      return;
+    }
+
+    if (layout) {
+      if (isCollapsed) {
+        this.setState({ isCollapsed: false, isResizable: true, size: { ...size, height: this._originalHeight } });
```

Comment:
We should be able to get originalHeight by just checking the height of all children. if it's in a collapsed state to start with (and height 1). _originalHeight will be set to 1 in the ctor so will set it to 1 again.